File: DIYHomeAutomation/app/src/main/java/com/example/diyhomeautomation/models/Restriction.kt
package com.example.diyhomeautomation.models

/**
 * Data class representing restrictions in the DIY Home Automation system.
 *
 * @property id The unique identifier for the restriction.
 * @property name The name of the restriction.
 * @property state The general state of the restriction.
 * @property condition The type of condition for the restriction.
 * @property primarySensorId The ID of the primary sensor associated with the restriction.
 * @property primarySensorState The general state of the primary sensor.
 * @property primarySensorValue The value of the primary sensor.
 * @property secondarySensorId The ID of the secondary sensor associated with the restriction.
 * @property secondarySensorState The general state of the secondary sensor.
 * @property secondarySensorValue The value of the secondary sensor (nullable).
 *
 * @constructor Creates an instance of Restriction with the provided properties.
 */
data class Restriction(
    var id: Int,
    var name: String,
    var state: GeneralState,
    var condition: TypeCondition,
    var primarySensorId: Int,
    var primarySensorState: GeneralState,
    var primarySensorValue: Double,
    var secondarySensorId: Int,
    var secondarySensorState: GeneralState,
    var secondarySensorValue: Double?
)

File: DIYHomeAutomationAPI/DIYHomeAutomationAPI/Models/Restriction.cs
using DIYHomeAutomationAPI.Models.Enums;

namespace DIYHomeAutomationAPI.Models
{
    /// <summary>
    /// This class represents a Restriction
    /// </summary>
    public class Restriction
    {

        /// <summary>
        /// Restriction's Id
        /// </summary>
        public int Id { get; set; }

        /// <summary>
        /// Restriction's Name
        /// </summary>
        public string Name { get; set; } = null!;

        /// <summary>
        /// Restriction's State
        /// </summary>
        public bool State { get; set; }

        /// <summary>
        /// Restriction's Condition
        /// </summary>
        public TypeCondition Condition { get; set; }

        /// <summary>
        /// Primary Device's Id
        /// </summary>
        public int PrimaryDeviceId { get; set; }

        /// <summary>
        /// Primary Device's State
        /// </summary>
        public bool PrimaryDeviceState { get; set; }

        /// <summary>
        /// Primary Device's Value
        /// </summary>
        public double PrimaryDeviceValue { get; set; }

        /// <summary>
        /// Secondary Device's Id
        /// </summary>
        public int? SecondaryDeviceId { get; set; }

        /// <summary>
        /// Secondary Device's State
        /// </summary>
        public bool? SecondaryDeviceState { get; set; }

        /// <summary>
        /// Secondary Device's Value
        /// </summary>
        public double? SecondaryDeviceValue { get; set; }
    }
}
